src/main/java/org/codeforamerica/shiba/pages/RoutingDecisionService.java

```diff
@@ -66,6 +67,30 @@ public RoutingDecisionService(CountyParser countyParser,
     return List.of(countyRoutingDestinations.get(county));
   }
 
+  private List<RoutingDestination> routeRedLakeClients(Set<String> programs,
+      ApplicationData applicationData, Document document, County county) {
+    if (!isLivingInTribalNationBoundary(applicationData)
+        || isOnlyApplyingForGrh(programs, applicationData)) {
+      return List.of(countyRoutingDestinations.get(county));
+    }
+
+    if (programs.contains(GRH)) {
+      return List.of(countyRoutingDestinations.get(county), tribalNations.get(RED_LAKE));
+    }
+
+    return List.of(tribalNations.get(RED_LAKE));
+  }
+
+  private boolean isLivingInTribalNationBoundary(ApplicationData applicationData) {
+    return Boolean.parseBoolean(
```

Comment:
Side thought - we could make this a convenient method in ApplicationDataParser "getBoolean(pagesData, field)"